tests/functional/adapter/materialized_view_tests/utils.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Optional
+
+from dbt.adapters.base.relation import BaseRelation
+from dbt.tests.util import run_dbt_and_capture
+
+from dbt.adapters.databricks.relation import DatabricksRelation
+
+
+def query_relation_type(project, relation: BaseRelation) -> Optional[str]:
+    assert isinstance(relation, DatabricksRelation)
+    sql = f"""
+    select
+        'table' as relation_type
+    from pg_tables
+    where schemaname = '{relation.schema}'
+    and tablename = '{relation.identifier}'
+    union all
+    select
+        case
+            when definition ilike '%create materialized view%'
+            then 'materialized_view'
+            else 'view'
+        end as relation_type
+    from pg_views
+    where schemaname = '{relation.schema}'
+    and viewname = '{relation.identifier}'
+    """
+    results = project.run_sql(sql, fetch="all")
```

Comment:
@mikealfare how can I change this to use `list_relations_without_caching()` instead? in my mind that's where this logic is already standardized.

in the case of dbt-databricks, it all the logic is in Python ( [`DatabricksAdapter.get_relations_without_caching()`](https://github.com/dataders/dbt-databricks/blob/539a40f2e9035386b8e96d341908b4795b5f6d43/dbt/adapters/databricks/impl.py#L221-L297). I'd rather not duplicate/reimplement it

Reply:
in the meantime, i suppose this is a decent woraround
```sql
    select
        'table' as relation_type
    from information_schema.tables
    where schemaname = '{relation.schema}'
    and tablename = '{relation.identifier}'
    union all
    select
        case when
            is_materialized = TRUE then 'materialized_view'
            else 'view'
        end as relation_type
    from information_schema.views
    where schemaname = '{relation.schema}'
    and viewname = '{relation.identifier}'
```

Reply:
So long as its only used with the UC cluster...information schema isn't available on the non-uc cluster.